src/timeout.ts

```diff
@@ -352,3 +352,20 @@ export class LegacyTimeoutContext extends TimeoutContext {
     return;
   }
 }
+
+/** @internal
+ * When `timeoutContext` is defined, throws a MongoOperationTimeoutError if the context has expired.
+ * If the context has not expired, returns the `remainingTimeMS`
+ * If the `timeoutContext` is not defined, returns undefined
+ * */
+export function getRemainingTimeMSOrThrow(
```

Comment:
Added this helper to prevent excessive duplication of code

Reply:
Don't feel strongly but I would've expected this to be a method on the class, and the null case can be handled by `?.`

Reply:
Just want your thoughts here in case this was missed